toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -300,6 +301,49 @@ func (g *PkgGraph) validateNodeForLookup(pkgNode *PkgNode) (valid bool, err erro
 	return
 }
 
+/*check if a runNode already exists for the same version of the remoteNode, if exists, replace runNode in the lookup with remoteNode*/
+func (g *PkgGraph) AddRemoteToLookup(pkgNode *PkgNode, deferSort bool) (err error) {
+	var (
+		requestInterval, nodeInterval pkgjson.PackageVerInterval
+	)
+
+	if pkgNode.Type != TypeRemoteRun {
+		logger.Log.Tracef("Skipping %+v, not valid for replacing lookup", pkgNode)
```

Comment:
I'm being cautious here. We always pass remote nodes only. Returning the error anyways, just in case.

Reply:
There is no guarantee for the error being returned from here will be handled as intended. So, panicking here itself to surface the issue instead of silently proceeding. With current code this scenario will never happen.